students/ThanhNhan/lesson07/html_render.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+import pytest
+"""
+A class-based system for rendering html.
+"""
+
+
+# This is the framework for the base class
+class Element:
+    tag = "html"
+    indent = "    "
+    def __init__(self, content=None, **kwargs):
+       
+        if content is None:
+            self.contents = []
+        else:
+            self.contents = [content]
+        self.attributes = kwargs
+        print("contents is", self.contents)
+        print("kwargs ", self.attributes)
+
+
+    def append(self, new_content):
+        self.contents.append(new_content)
+
+    def render(self, out_file, cur_ind=""):   
+       # self.indent = cur_ind
+        out_file.write(cur_ind + self._open_tag())        
+       # out_file.write("<{}>\n".format(self.tag))
+        for line in self.contents:
+            try:
+                line.render(out_file, cur_ind + self.indent)
+            except AttributeError:
+                out_file.write(cur_ind + self.indent + line)
+                out_file.write("\n")
+        out_file.write(cur_ind + self._close_tag())
+
+    def _open_tag(self):
+        open_tag = []
+        if (self.attributes is None):
```

Comment:
will `self.attributes` ever be None? I think it will always exist -- if not attributes, it will be an empty dict.

Reply:
I just removed this line and all tests still passed.